sql/core/src/test/resources/sql-tests/results/sql-compatibility-functions.sql.out

```diff
@@ -102,7 +102,7 @@ true	1	1	1	1
 -- !query 10
 SELECT float(1), double(1), decimal(1)
 -- !query 10 schema
-struct<CAST(1 AS FLOAT):float,CAST(1 AS DOUBLE):double,CAST(1 AS DECIMAL(10,0)):decimal(10,0)>
+struct<CAST(1 AS FLOAT):float,CAST(1 AS DOUBLE):double,CAST(1 AS DECIMAL(100)):decimal(10,0)>
```

Comment:
Hi, @gatorsmile . The PR looks good. BTW, could we make `DECIMAL(10 0)` instead of `DECIMAL(100)` by changing [DecimalType](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala#L66) together in this PR?
```
-  override def typeName: String = s"decimal($precision,$scale)"
+  override def typeName: String = s"decimal($precision, $scale)"
```